service/src/main/java/io/spaship/operator/event/EventSourcingEngine.java

```diff
@@ -0,0 +1,64 @@
+package io.spaship.operator.event;
+
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.JsonObject;
+import io.vertx.mutiny.core.Vertx;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Singleton;
+import java.time.LocalDateTime;
+import java.util.Optional;
+import java.util.UUID;
+
+@Singleton
```

Comment:
@ApplicationScoped is commonly used in Quarkus which is at the end Singleton

Reply:
ok I'm fine with @Singleton if you think it's a proper usage. You're right about proxying but it's useful for testing resp. mocking.